azurelinuxagent/common/protocol/hostplugin.py

```diff
@@ -65,16 +65,20 @@ class HostPluginProtocol(object):
     FETCH_REPORTING_PERIOD = datetime.timedelta(minutes=1)
     STATUS_REPORTING_PERIOD = datetime.timedelta(minutes=1)
 
-    def __init__(self, endpoint, container_id, role_config_name):
+    def __init__(self, endpoint):
+        """
+        NOTE: Before using the HostGAPlugin be sure to invoke GoalState.update_host_plugin_headers() to initialize
+              the container id and role config name
+        """
         if endpoint is None:
             raise ProtocolError("HostGAPlugin: Endpoint not provided")
         self.is_initialized = False
         self.is_available = False
         self.api_versions = None
         self.endpoint = endpoint
-        self.container_id = container_id
-        self.deployment_id = self._extract_deployment_id(role_config_name)
-        self.role_config_name = role_config_name
+        self.container_id = None
+        self.deployment_id = None
+        self.role_config_name = None
```

Comment:
Now these properties need to be explicitly initialized by calling GoalState.update_host_plugin_headers(). 

This is just a temporary change; I need to do some refactoring in the initialization of GoalState/WireProtocol/HostGAPluginProtocol/Telemetry and will improve on this.